mqtt/mqtt-broker/src/broker.rs

```diff
@@ -38,6 +38,8 @@ where
     retained: HashMap<String, proto::Publication>,
     authenticator: N,
     authorizer: Z,
+
+    pub on_publish: Option<Sender<()>>,
```

Comment:
q: I'm not sure, but is it possible to guard this with `[cfg(test)]`?

Reply:
not with tests, I'm going to add special feature and run benches with this feature enabled